src/coreclr/md/enc/metamodelrw.cpp

```diff
@@ -7144,8 +7144,10 @@ CMiniMdRW::ValidateVirtualSortAfterAddRecord(
 void
 CMiniMdRW::Debug_CheckIsLockedForWrite()
 {
+#ifdef HOST_WINDOWS
```

Comment:
This is not an appropriate limitation. There should be nothing Windows specific about this code path.

Reply:
I can't find a way to test whether a `pthread_rwlock_t` is held.